plugins/newsdownloader.koplugin/epubdownloadbackend.lua

```diff
@@ -24,6 +24,47 @@ local max_redirects = 5; --prevent infinite redirects
 local TIMEOUT_CODE = "timeout" -- from socket.lua
 local MAXTIME_CODE = "maxtime reached" -- from sink_table_with_maxtime
 
+local function filter(text, element)
+    local htmlparser = require("htmlparser")
+    local root = htmlparser.parse(text)
+    local filtered = ""
+    
+    local selectors = {
+        "main",
+        "article",
+        "div#main",
+        "#main-article",
+        ".main-content",
+        "#body",
+        "#content",
+        ".content",
+        "div#article",
+        "div.article",
+        "div.post",
+        "div.post-outer",
+        ".l-root",
+        ".content-container",
+        ".StandardArticleBody_body",
+        "div#article-inner",
+        "div#newsstorytext",
+        "div.general",
+        }
+    
+    if element then selectors[0] = element end
+    
+    for _,sel in pairs(selectors) do
+       local elements = root:select(sel)
+       if not elements then return text end
```

Comment:
Is it right that you return when the first selector gives no element ? Shouldn't you try next selectors?
(There's no `continue` in Lua, only `break` so you have to imbricate all positive checks and let it continue naturally when it reaches the end of the `for` .)

Reply:
Thanks, I did not notice.